### PR TITLE
scripts/vsmlrt.py: add support for esrgan janai models

### PR DESCRIPTION
**Pull request by hooke007**
Continue from https://github.com/AmusementClub/vs-mlrt/issues/41#issuecomment-1511414723

edit:
[models - 1.zip](https://github.com/AmusementClub/vs-mlrt/files/11280019/models.-.1.zip)

### CODE REVIEW

**Comment by AkarinVS** [comment]
there will be more and more waifu2x/esrgan/cugan-like models in the future, and it's impractical to maintaining an up-to-date support of all such models in the python wrapper, at least not in a timely manner. so my questions are:
1. what should be criteria for adding one? and also for including in the official builtin models package?
2. could we design a generic python API (the current `vsmlrt.inference` is one such design, but obviously it fails to attract usage) to support using those models without having the users waiting for the `vsmlrt.py` to update?

**Comment by hooke007** [comment]
>  and it's impractical to maintaining an up-to-date support of all such models in the python wrapper, at least not in a timely manner.

Agreed, actually this is a draft. I prefer to only keep two or three models of janai (one for speed and one for quality). Comparing the previous esrgan's official model. janai could preserve anime's original art style rather than totally rebuilding(/breaking?) it. Of course, from the perspective of serious encoding, there is still a gap in its level. This is what I think is lacking in the current ESRGAN model.

**Comment by AkarinVS** [comment]
After extensive discussions, we propose the following policy regarding contributing models into vs-mlrt:
1. we will add a new model package (besides the existing `models.7z` and `ext-models.7z`), `contrib-models.7z` to host community contributed onnx files to our releases.
2. The python wrappers, depending on whether it needs to reuse existing model specific functions in `vsmlrt.py`, live in either of these files (both files will be included in main release, only the onnx files are packaged separately due to space constraints):
   - if it has to modify the existing code, like in this PR's case, you can modify `vsmlrt.py`, but please make the model id number starting from 5000, so that the code can know if the corresponding ONNX file is coming from `models`/`ext-models` or `contrib-models`. If you modify the `vsmlrt.py`, please make sure any code modifications has properly isolated from the rest of the code (to avoid breaking existing models).
   - If it's an entirely new model family that you have to write your own model specific functions (e.g. `vsmlrt.Waifu2x`), then you can add such code into a new `contrib.py` file (specifics to be released later after we've done some reorganizations).
3. for a PR to be accepted, the author *must* agree to maintain the ONNX model files and corresponding python wrappers. If a contributed model is broken for some time and its original author fails to respond to issue reports, we reserve the rights to remove such models.
4. we will add a rolling nightly prerelease so that users can immediately use contributed models without waiting for the next release.

What do you think about this proposed policy?

Thanks.

**Comment by hooke007** [comment]
I agree with this policy.

**Comment by hooke007** [comment]
> make the model id number starting from 5000

It would make out of tuple' range.

File: scripts/vsmlrt.py
Comment (WolframRhodium) [inline scripts/vsmlrt.py:479]: should not create new global variable

File: scripts/vsmlrt.py
Comment (WolframRhodium) [inline scripts/vsmlrt.py:545]:
```python3
f"{tuple(RealESRGANv2Model.__members__)[model]}.onnx".replace('_', '-')
```
is less verbose

Comment (hooke007) [reply scripts/vsmlrt.py:545]: https://github.com/AmusementClub/vs-mlrt/pull/42#issuecomment-1515751375

File: scripts/vsmlrt.py
Comment (WolframRhodium) [inline scripts/vsmlrt.py:470]: should be labeled as contributed models, and add the link to the original repository and maintainer

File: scripts/vsmlrt.py
Comment (WolframRhodium) [inline scripts/vsmlrt.py:492]: probably just use
```python3
model: RealESRGANv2Model = 0
```
to simplify typing

File: scripts/vsmlrt.py
Comment (WolframRhodium) [inline scripts/vsmlrt.py:509]:
```python3
raise ValueError(f'{func_name}: invalid "model"')
```
otherwise it looks a bit verbose

**Comment by hooke007** [comment]
@WolframRhodium still out of tuple's range
Snipaste_2023-04-20_09-45-50

**Comment by WolframRhodium** [comment]
That's a better solution. LGTM. Please also attach your name.

**Comment by hooke007** [comment]
> Please also attach your name.

Hmm, Where should I add into the code?

**Review by WolframRhodium — APPROVED** [review]

File: scripts/vsmlrt.py
Comment (hooke007) [reply scripts/vsmlrt.py]: Sry, I'm confused that github showed the outdated msg?

Snipaste_2023-04-25_02-10-41

Do I need to change the current commit? https://github.com/AmusementClub/vs-mlrt/pull/42/commits/f0ec6ae322c742dca024e76d0f582514cf68f7f0

Comment (WolframRhodium) [reply scripts/vsmlrt.py]: nothing, thanks for your contribution!

**Pull request merged**
